src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -132,9 +126,16 @@ public override string Id
             {
                 if (_lazyId == null)
                 {
-                    _lazyId = Path.GetFileName(this.FullPath).ToLower();
+                    try
+                    {
```

Comment:
why are we using name rather than mvid or combination of name + mvid?

name seems too easy to be collide